### PR TITLE
broadcast block to dpos network when received higher block

### PR DESCRIPTION
**Pull request by RainFallsSilent**

### CODE REVIEW

File: dpos/manager/dposmanager.go [hunk 1]
```diff
@@ -114,6 +120,8 @@ type DPOSManager struct {
 	recoverStarted       bool
 	notHandledProposal   map[string]struct{}
 	statusMap            map[uint32]map[string]*dmsg.ConsensusStatus
+
```
Comment (alexpan0610) [inline dpos/manager/dposmanager.go]: Remove blank line.

File: dpos/manager/dposmanager.go [hunk 2]
```diff
@@ -153,6 +161,7 @@ func (d *DPOSManager) Initialize(handler *DPOSHandlerSwitch,
 	d.broadcast = broadcast
 	d.notHandledProposal = make(map[string]struct{})
```
Comment (alexpan0610) [inline dpos/manager/dposmanager.go]: Line 162~164, these parameters should initiate in New() method.

File: dpos/manager/dposmanager.go [hunk 3]
```diff
@@ -267,6 +278,7 @@ func (d *DPOSManager) OnPong(id dpeer.PID, height uint32) {
 
 func (d *DPOSManager) OnBlock(id dpeer.PID, block *types.Block) {
 	log.Info("[ProcessBlock] received block:", block.Hash().String())
+	delete(d.requestedBlocks, block.Hash())
```
Comment (alexpan0610) [inline dpos/manager/dposmanager.go]: Show warn log if received block not in requested list.

File: dpos/manager/dposmanager.go (hunk 1 above)
Comment (RainFallsSilent) [reply dpos/manager/dposmanager.go]: "neededMajorityStatus" "recoverStarted" "notHandledProposal" "statusMap" is for recover abnormal state,  requestedBlocks if for broadcast block for dpos network.  “requestedBlocks” should be separated.

File: dpos/manager/dposmanager.go (hunk 2 above)
Comment (RainFallsSilent) [reply dpos/manager/dposmanager.go]: fixed

File: dpos/manager/dposmanager.go (hunk 3 above)
Comment (RainFallsSilent) [reply dpos/manager/dposmanager.go]: fixed

**Pull request merged**